homeassistant/components/myuplink/sensor.py

```diff
@@ -0,0 +1,126 @@
+"""Sensor for myUplink."""
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from myuplink.models import DevicePoint
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from . import MyUplinkDataCoordinator
+from .const import DOMAIN
+from .coordinator import CoordinatorData
+from .entity import MyUplinkEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class MyUplinkDeviceSensorEntityDescription(SensorEntityDescription):
+    """Describes MyUplink device sensor entity."""
+
+    value_fn: Callable[[CoordinatorData, str], StateType]
+
+
+DEVICE_POINT_DESCRIPTIONS = {
+    "°C": SensorEntityDescription(
+        key="data_point",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up myUplink sensor."""
+    entities: list[SensorEntity] = []
+    coordinator: MyUplinkDataCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    # Setup device point sensors
+    for device_id, point_data in coordinator.data.points.items():
+        for point_id, device_point in point_data.items():
+            entities.append(
+                MyUplinkDevicePointSensor(
+                    coordinator=coordinator,
+                    device_id=device_id,
+                    device_point=device_point,
+                    entity_description=DEVICE_POINT_DESCRIPTIONS.get(
+                        device_point.parameter_unit
+                    ),
+                    unique_id_suffix=point_id,
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class MyUplinkDeviceSensor(MyUplinkEntity, SensorEntity):
```

Comment:
Let's remove this class and the entity description class that aren't used now. If we need it later we can just look at the history of this PR and copy it back.